### PR TITLE
#165273469 Add jwt authenticatication

### PR DESCRIPTION
**Pull request by engjames**
**What does this PR do?**
-Add a jwt authentication

**Description of Task to be completed?**
- Add PyJWT to requirements.txt file
- Enable jwt authentication in settings.py
- Modify urls to cater for jwt authentication
- Add a method in the models that generates a token
- Add a method to check and verify for authorization header
- Return a response with a token, message and username upon successful generation.
For a person to get a token, he has to first create an account using http://127.0.0.1:8000/api/users/
**The body should be in the form**
  {
	"user": {
        "username": "francis",
        "email": "francis@gmail.com",
        "password": "123456789"
    }
}
Then login to get a token using http://127.0.0.1:8000/api/users/login/
**The body should be in the form**
{
	"user":{
	"email": "francis@gmail.com",
	"password": "123456789"
	}
}
image

**What are the relevant pivotal tracker stories?**
#165273469

@joelmugaya @Baalmart @mwinel @3Nakajugo @sanya-kenneth @KabohaJeanMark @archbishop12 @

### CODE REVIEW

File: authors/apps/authentication/backends.py [hunk 1]
```diff
@@ -1,10 +1,58 @@
...
+            payload = jwt.decode(token, settings.SECRET_KEY, 'utf-8')
+        except BaseException:
+            message = "The token provided can not be decoded."
+            raise exceptions.AuthenticationFailed(message)
+        user = User.objects.get(username=payload['sub']['username'])
+        if not user:
+            message = "User does not exist in the database."
+            raise exceptions.AuthenticationFailed(message)
+        if not user.is_active:
+            message = "User is deactivated."
```
Comment (joelmugaya) [inline authors/apps/authentication/backends.py]: Return an appropriate message. eg. 
```suggestion
            message = "User is not active."
```

File: authors/apps/authentication/serializers.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
+"""Imports relevant to our class."""
```
Comment (joelmugaya) [inline authors/apps/authentication/serializers.py]: Thanks for adding a comment but try and avoid unnecessary changes

File: authors/apps/authentication/urls.py [hunk 3]
```diff
@@ -1,3 +1,4 @@
+"""Import relevant dependencies."""
```
Comment (joelmugaya) [inline authors/apps/authentication/urls.py]: Thanks for adding a comment but try and avoid unnecessary changes

File: requirements.txt [hunk 4]
```diff
@@ -3,7 +3,6 @@ attrs==19.1.0
 certifi==2019.3.9
 chardet==3.0.4
 coverage==4.5.3
-coveralls==1.7.0
```
Comment (joelmugaya) [inline requirements.txt]: We need this package please fix

**Review by archibishop — APPROVED** [review]
LGMT @engjames  I am merging this. Good work.
